packages/calling-stateful-client/src/StatefulCallClient.ts

```diff
@@ -133,6 +133,31 @@ export interface StatefulCallClient extends CallClient {
     participantId: CommunicationIdentifier | undefined,
     stream: LocalVideoStreamState | RemoteVideoStreamState
   ): void;
+  /**
+   * Update the scaling mode of a {@link VideoStreamRendererView}
+   * {@link @azure/communication-calling#VideoStreamRendererView.updateScalingMode}.
+   *
+   * Its important to updateViewScalingMode correctly updates the appropriate views.
+   *
+   * Scenario 1: updateViewScalingMode for a VideoStreamRendererView of a Remote Participant
+   * - CallId is required, participantId is required, and @JamesBurnside TODO: SOMETHING
+   *
+   * Scenario 2: updateViewScalingMode of a local VideoStreamRendererView in a call
+   * - CallId is required, participantId must be undefined, and @JamesBurnside TODO: SOMETHING
+   *
+   * - Scenario 2: updateViewScalingMode od a local VideoStreamRendererView not part of a call
+   * - CallId must be undefined, participantId must be undefined, and @JamesBurnside TODO: SOMETHING
+   *
+   * @param callId - CallId for the given stream. Can be undefined if the stream is not part of any call.
+   * @param participantId - {@link RemoteParticipant.identifier} associated with the given RemoteVideoStreamState. Could
+   *   be undefined if disposing LocalVideoStreamState.
+   * @param stream - The LocalVideoStreamState or RemoteVideoStreamState to dispose.
+   */
+  updateViewScalingMode(
```

Comment:
I think it was a mistake to have "polymorphic" `createView` and `disposeView` where you get different behaviors based on some combinations of the arguments.

Let's rethink this API surface if we're adding more of the kind.